### ISSUE TITLE
Allow teams to require 2fa for users

### ISSUE THREAD

**Issue by snopoke** [dimagi/open-chat-studio#147]
This should be quite easy with https://django-allauth-2fa.readthedocs.io/en/latest/advanced/

**Comment by Davemaina1** [comment]
.take

**Comment by Davemaina1** [comment]
**Problem:** MFA enforcement (`RequireMfaForStaffMiddleware`) is global, staff-only, and gated by `REQUIRE_MFA_FOR_STAFF`. Teams have no way to require it for their own members.

**Approach:** Extend the middleware rather than add a second one. Rename it to `RequireMfaMiddleware` (it stops being staff-only) and change its enrollment check to "staff/superuser gate (unchanged) OR the user belongs to a team requiring MFA." The team side reuses `apps/teams/flags.py`'s flag system: a new `flag_require_mfa` entry with `teams_can_manage=True` puts a checkbox on the team settings page (`apps/teams/views/feature_flags.py`), gated to team admins by that view's existing permission check. No new migration, admin form, or template.

A `Team.require_mfa` boolean would need its own migration, admin form, and settings section — a second implementation of what the flag system (team-scoped, admin-toggleable, cached, ADR-0058) already does.

`django-allauth-2fa`, linked from the issue, isn't an option: the repo runs allauth's own `mfa` app (`django-allauth[mfa]`), and the two don't run together. Its one useful idea, the `BaseRequire2FAMiddleware` pattern, is what `RequireMfaForStaffMiddleware`'s docstring already says it follows.

**Scope:** every member of a flagged team is enrolled, not just admins — matches the issue title. A user in two teams where only one requires MFA is gated everywhere, not just on that team's pages: same all-or-nothing shape as the staff gate, and simpler than resolving "which team is this request for" on every path.

The check has to look across all of a user's teams, not `request.team`: `TeamsMiddleware` sets `request.team` inside its own `process_view`, and `process_view` hooks run in `MIDDLEWARE` order — `RequireMfaForStaffMiddleware` is listed above `TeamsMiddleware`, so it runs first, before `request.team` exists.

**Blast radius:** `apps/users/middleware.py` (renamed), `config/settings.py`, and the middleware's test file change; one entry is added to `apps/teams/flags.py`. `apps/teams/models.py`, `Flag`, and the settings-page view/form need no changes — confirmed against `apps/teams/tests/test_flags.py`, which tests the flag mechanism generically, not a fixed flag list. `REQUIRE_MFA_FOR_STAFF`'s own tests are unaffected.

**Test plan:** extend `apps/users/tests/test_require_mfa_middleware.py` with the same `TeamWithUsersFactory` and real-login pattern. New cases: a non-staff member of a flagged team is sent to MFA setup; a member of an unflagged team isn't; a user in two teams where only one is flagged is still gated. Existing staff-only cases stay green. The settings-page toggle needs no new test — existing flag tests already cover any new entry.

**Delivery:** one PR against `main`, closing #147.

**Comment by snopoke** [comment]
> The team side reuses apps/teams/flags.py's flag system: a new flag_require_mfa entry with teams_can_manage=True puts a checkbox on the team settings page (apps/teams/views/feature_flags.py), gated to team admins by that view's existing permission check. No new migration, admin form, or template.

We shouldn't use flags for this. Flags are intended to feature rollout and not for settings. Prefer a new audited field on the team.

**Comment by Davemaina1** [comment]
Agreed, flags are for rollout, not settings. Switched to an audited `Team.require_mfa` boolean: its own migration, a `TeamMfaForm` on the team settings page (same pattern as `is_migrating`), `require_mfa` added to `TEAM_FIELDS` for `field_audit`. Middleware now checks `user.teams.filter(require_mfa=True).exists()` instead of the flag.
